drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -67,13 +68,25 @@ static int stm32_temp_channel_get(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32L5X
+	BUILD_ASSERT(TEMPSENSOR_CAL_VREFANALOG != 0);
+	temp = ((float)data->raw * cfg->tsv_mv) / TEMPSENSOR_CAL_VREFANALOG;
+	temp -= *TEMPSENSOR_CAL1_ADDR;
+	temp *= (TEMPSENSOR_CAL2_TEMP - TEMPSENSOR_CAL1_TEMP);
+	temp /= (*TEMPSENSOR_CAL2_ADDR - *TEMPSENSOR_CAL1_ADDR);
+	temp += TEMPSENSOR_CAL1_TEMP; /* temperature offset: 30°C */
+#else
+	int32_t mv = data->raw * cfg->tsv_mv / 0x0FFF; /* Sensor value in millivolts */
+
 	if (cfg->is_ntc) {
-		temp = (float)(cfg->v25_mv - data->mv);
+		temp = (float)(cfg->v25_mv - mv);
 	} else {
-		temp = (float)(data->mv - cfg->v25_mv);
+		temp = (float)(mv - cfg->v25_mv);
 	}
-	temp = (temp/cfg->avgslope)*10;
+	temp = (temp / cfg->avgslope) * 10;
 	temp += 25;
```

Comment:
Modifications to existing code should be done in a prelinary commit (so it is easier to review that you're not breaking anything)

Reply:
done